### PR TITLE
LRDOCS-8203 Incorporating liferay-npm-scripts into Learn

### PR DESCRIPTION
**Pull request by rdai10**
@brianchandotcom :
Resend per https://github.com/brianchandotcom/liferay-learn/pull/68

This pr includes changes for:
- Integrating with liferay-npm-scripts for formatting the code. 
I ran into a few issues. Such as formatting js files will throw errors because liferay-npm-scripts does not recognize jquery's `$`. And issue with having the tool format md files in the docs directory. I created an open issue at https://github.com/liferay/liferay-frontend-projects/issues/91.
- Added a github action to check format.
- A minor bug fix to align search icon on smaller mobile devices.
- Added ability to expand and collapse the article navs on mobile.
- Minified css.
- Updated `.yml` file to use 4 spaces instead of 2.

Let me know if you have any questions, thanks!
cc: @jrhoun

### CODE REVIEW

**Comment by rdai10** [comment]
Looks like for github actions, the steps have to line up (2 space indents), otherwise it throws an error:
image

File: site/homepage/package.json [hunk 1]
```diff
@@ -33,8 +33,8 @@
 		"build": "npm run format && npm run build-css && npm run propagate-img && npm run propagate-js && npm run propagate-template",
 		"build-css": "node-sass --include-path node_modules/clay-css/src/scss _static/scss/main.scss _static/main.css && cp _static/main.css ../docs/_static",
 		"css:watch": "node-sass -w --include-path node_modules/clay-css/src/scss _static/scss/main.scss _static/main.css",
-		"format": "prettier --write .",
-		"format-check": "prettier --check .",
+		"format": "liferay-npm-scripts fix",
+		"format-check": "yarn liferay-npm-scripts check",
```
Comment (wincent) [inline site/homepage/package.json:35]: You don't need the `yarn` here.

You also have a couple of options if you want to roll this out more "gently".

- One is to use `fix` and `check` like you are here. Those are going to run _all_ of ESLint, Prettier, stylelint, plus any additional "pre-flight" checks that `@liferay/npm-scripts` implements. This is obviously a higher barrier which may add some friction in return for more consistency/quality (at least, that's the theory).
- The other is to use `format` and `format:check` (ie. `liferay-npm-scripts format` and `liferay-npm-scripts format:check`). This _only_ runs Prettier, so the barrier to entry is lower and it is easier to turn on this lighter-weight check, and later on tighten things up by activating the broader check.

Comment (rdai10) [reply site/homepage/package.json]: Thanks @wincent for the suggestion!
6161ded

**Comment by rdai10** [comment]
For reference, @wincent created https://github.com/liferay/liferay-frontend-projects/issues/92 to allow npm-scripts to format .md and .yml files.

**Comment by brianchandotcom** [comment]
Merged. Thank you.
View total diff: https://github.com/brianchandotcom/liferay-learn/compare/949f997b02...1aa051bd9c

**Comment by brianchandotcom** [comment]
@rdai10 see 1aa051bd9c4fd2dbcad3071cacc4e7d91ad876b6

very nice @wincent @rdai10 !

**Comment by rdai10** [comment]
Following up with https://github.com/brianchandotcom/liferay-learn/pull/70

**Pull request closed without merging**